garage/experiment/local_tf_runner.py

```diff
@@ -163,41 +169,132 @@ def obtain_samples(self, itr, batch_size):
+                See examples below.
+
+        Returns:
+            None if resume_now=True.
+            Otherwise a dict of arguments for train().
+
+        Examples:
+            1. Resume via command line. (Recommended)
+            ./examples/resume_training.py --snapshot_dir /saved/dir
+
+            2. Resume experiment immediately.
+            with LocalRunner() as runner:
+                runner.restore(snapshot_dir)
+
+            3. Resume experiment with modified training arguments.
+             with LocalRunner() as runner:
+                args = runner.restore(snapshot_dir, resume_now=False)
+                args['n_epochs'] = 20
+                runner.train(**args)
+
+        Note:
+            When resume via command line, new snapshots will be
+            saved into the SAME directory if not specified.
+
+            When resume programmatically, snapshot directory should be
+            specify manually or through run_experiment() interface.
+        """
+        snapshotter.snapshot_dir = snapshot_dir
+        saved = snapshotter.load(from_epoch)
+
+        self.setup_args = saved['setup_args']
+        self.train_args = saved['train_args']
+
+        self.setup(
+            env=saved['env'],
+            algo=saved['algo'],
+            sampler_cls=self.setup_args['sampler_cls'],
+            sampler_args=self.setup_args['sampler_args'])
+
+        n_epochs = self.train_args['n_epochs']
+        last_epoch = saved['last_epoch']
+        n_epoch_cycles = self.train_args['n_epoch_cycles']
+        batch_size = self.train_args['batch_size']
+        store_paths = self.train_args['store_paths']
+        plot = self.train_args['plot']
+        pause_for_plot = self.train_args['pause_for_plot']
+
+        logger.log('Restore from snapshot saved in %s' % snapshot_dir)
+        logger.log('{:<20} {:<15}'.format('Train Args', 'Value'))
+        logger.log('{:<20} {:<15}'.format('n_epochs', n_epochs))
+        logger.log('{:<20} {:<15}'.format('last_epoch', last_epoch))
+        logger.log('{:<20} {:<15}'.format('n_epoch_cycles', n_epoch_cycles))
+        logger.log('{:<20} {:<15}'.format('batch_size', batch_size))
+        logger.log('{:<20} {:<15}'.format('store_paths', store_paths))
+        logger.log('{:<20} {:<15}'.format('pause_for_plot', pause_for_plot))
+
+        if resume_now:
+            return self._train(
+                n_epochs,
+                start_epoch=last_epoch + 1,
+                n_epoch_cycles=n_epoch_cycles,
+                batch_size=batch_size,
+                store_paths=store_paths,
+                plot=plot,
+                pause_for_plot=pause_for_plot)
+        else:
+            args = self.train_args.copy()
+            args.update(start_epoch=last_epoch + 1)
+            return args
+
     def log_diagnostics(self, pause_for_plot=False):
         """Log diagnostics.
 
         Args:
-            pause_for_plot: Pause for plot.
+            pause_for_plot(bool): Pause for plot.
```

Comment:
we don't actually use the `log_diagnostics` API anymore, so feel free to just log wherever you see fit.